spec/factories/miq_request_task.rb

```diff
@@ -38,12 +38,22 @@
     state        'pending'
     request_type 'clone_to_service'
   end
-  factory :service_retire_task,             :parent => :miq_retire_task,  :class => "ServiceRetireTask" do
-    request_type 'service_retire'
-    state        'pending'
-  end
 
   factory :service_template_transformation_plan_task, :parent => :service_template_provision_task, :class => 'ServiceTemplateTransformationPlanTask' do
     request_type 'transformation_plan'
   end
+
+  # Retire Tasks
+  factory :service_retire_task,             :parent => :miq_retire_task, :class => "ServiceRetireTask" do
+    request_type 'service_retire'
+    state        'pending'
+  end
+  factory :vm_retire_task,                  :parent => :miq_retire_task, :class => "VmRetireTask" do
+    request_type 'vm_retire'
```

Comment:
Feels like this is a model problem.  There's no other valid value, right?

Reply:
Sure, that's kinda what I thought too, but https://travis-ci.org/ManageIQ/manageiq/jobs/441749019#L1568 ...